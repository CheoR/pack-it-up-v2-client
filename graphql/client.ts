import { ApolloClient } from "@apollo/client"

import { getHost } from "./getHost"
import { cache } from "./cache"

export const client = new ApolloClient({
  // with expo the following do not work
  // "http://localhost:4000/"
  // "exp://192.168.1.65:19000/"
  // "http://localhost:4000/graphql"
  // "exp://192.168.1.65:19000/graphql"
  uri: getHost(),
  cache,
  credentials: "include",
  defaultOptions: { watchQuery: { fetchPolicy: "cache-and-network" } },
})
