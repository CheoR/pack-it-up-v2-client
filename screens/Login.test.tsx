import React from "react";
import { render, screen, fireEvent } from "@testing-library/react-native";
import { MockedProvider } from "@apollo/client/testing";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";

import LoginScreen, { LOGIN_USER } from "./Login";

const validLoginResp = {
  data: {
    loginUser: {
      accessToken: "accessTokenResp",
      refreshToken: "refreshTokenResp",
      user_id: "userId",
    },
  },
};

const mocks = [
  {
    request: {
      query: LOGIN_USER,
      variables: {
        email: "user@email.com",
        password: "userpassword",
      },
    },
    result: validLoginResp,
  },
];

describe("<LoginScreen />", () => {
  const mockFn = jest.fn();

  // it("responds with user login", async () => {
  it("fetches login button", async () => {
    const { debug, getByTestId, getByText } = render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <LoginScreen />
      </MockedProvider>
    );
    const btn = getByTestId("login");
    console.log(btn);
    expect(btn).toBeDefined();

    // <MockedProvider mocks={mocks} addTypename={false}>
    //   render(
    //   <LoginScreen />
    //   ); const resp = screen.getAllByLabelText("Login"); console.log(` ====
    //   resp`); console.log(resp); console.log(`==== `);
    //   expect(resp).toInclude("login");
    // </MockedProvider>;

    // const tree = renderer.create(
    //   <MockedProvider mocks={mocks} addTypename={false}>
    //     <LoginScreen />
    //   </MockedProvider>
    // );
    // console.log(tree.toJSON());
    // expect(tree.toJSON()).toMatchSnapshot();
  });
});
