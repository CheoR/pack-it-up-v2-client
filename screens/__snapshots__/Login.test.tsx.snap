// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginScreen /> responds with user login 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F2F3EF",
      "flex": 1,
      "justifyContent": "space-between",
      "marginTop": 54,
      "paddingHorizontal": 8,
    }
  }
>
  <Text />
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "height": 100,
          "justifyContent": "space-between",
          "width": "80%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Email"
        style={
          {
            "borderBottomColor": "#00171F",
            "borderBottomWidth": 8,
          }
        }
      />
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
        style={
          {
            "borderBottomColor": "#00171F",
            "borderBottomWidth": 8,
          }
        }
      />
    </View>
    <View
      style={
        {
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                {
                  "color": "#00171F",
                },
              ]
            }
          >
            Login
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
          }
        }
      >
        <Text>
          No Account,
           
          <Text
            onPress={[Function]}
            style={
              {
                "color": "#FF6700",
              }
            }
          >
            Register!
          </Text>
        </Text>
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-evenly",
          "paddingHorizontal": 10,
          "paddingVertical": 15,
          "width": "50%",
        }
      }
    >
      <Text />
      <Text />
      <Text />
      <Text />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingVertical": 15,
        }
      }
    >
      <Text />
      <Text>
        2022
      </Text>
    </View>
  </View>
</View>
`;
